### DISCUSSION TITLE
ERROR foundry_compilers_artifacts_solc::sources

### DISCUSSION THREAD

**Discussion by Not-Sarthak** [Cyfrin/foundry-full-course-cu#2167] [Q&A] [answered]
Error:
```
2024-08-03T18:35:55.057225Z ERROR foundry_compilers_artifacts_solc::sources: error="/home/sarthak/solidity/foundry-fund-me/test/Counter.t.sol": No such file or directory (os error 2)
Error: 
failed to read artifact source file for `/home/sarthak/solidity/foundry-fund-me/test/Counter.t.sol`

Context:
- "/home/sarthak/solidity/foundry-fund-me/test/Counter.t.sol": No such file or directory (os error 2)
```

I am running into this error while trying to run `forge script script/DeployFundMe.s.sol`

I do not require Counter.sol anymore and hence removed it, but on running forge script command, it still tries to look out for both `Counter.sol` and `Counter.t.sol`.  Why?

Moreover on running the same command after adding empty Counter.sol and Counter.t.sol, the script runs but deploys empty Counter contract. 

```DeployFundMe.s.sol
//SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Test, console} from "forge-std/Test.sol";
import { FundMe } from "../src/FundMe.sol";

contract FundMeTest is Test{

    FundMe fundMe;

    function setUp() external {
        fundMe = new FundMe();
    }

    function testMinimumDollarIsFive() public view {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }

    function testOwnerIsMsgSender() public view {
        console.log(fundMe.i_owner());
        assertEq(fundMe.i_owner(), address(this));
    }
}```

**Comment by EngrPips** [comment, marked as answer]
Hello @Not-Sarthak, delete the counter file, run forge clean, and try running your script again. Let's see if the issue persists.

**Reply by nishantguvvada** [reply]
Thanks @EngrPips "forge clean" cleared the cache folder, compiled again and it worked!

**Reply by EngrPips** [reply]
Awesome.

**Reply by Not-Sarthak** [reply]
Yes this worked! Thanks again @EngrPips .

**Reply by EngrPips** [reply]
You are welcome, and You should mark as answered so it shows that this discussion got answered @Not-Sarthak

**Reply by yawarasuuna** [reply]
You save my life daily, you always have the right answer lol. Tyvm

**Reply by EngrPips** [reply]
> You save my life daily, you always have the right answer lol. Tyvm

Anytime, Friend. Keep crushing it.
